lib/exprengine.cpp

```diff
@@ -2646,19 +2647,41 @@ static std::string execute(const Token *start, const Token *end, Data &data)
                         data.assignStructMember(tok2, &*structVal, memberName, memberValue);
                         continue;
                     }
+
+                    // Assign a pointer value
                     if (lhs->isUnaryOp("*") && lhs->astOperand1()->varId()) {
                         const Token *varToken = tok2->astOperand1()->astOperand1();
+                        // Get value of the pointer
                         ExprEngine::ValuePtr val = data.getValue(varToken->varId(), varToken->valueType(), varToken);
-                        if (val && val->type == ExprEngine::ValueType::ArrayValue) {
-                            // Try to assign "any" value
-                            auto arrayValue = std::dynamic_pointer_cast<ExprEngine::ArrayValue>(val);
-                            arrayValue->assign(std::make_shared<ExprEngine::IntRange>("0", 0, 0), std::make_shared<ExprEngine::BailoutValue>());
+                        if (!val || (val->type != ExprEngine::ValueType::ArrayValue &&
+                                     val->type != ExprEngine::ValueType::AddressOfValue))
+                            throw ExprEngineException(tok2, "Unhandled assignment in loop");
+                        // Handle the value that we are pointing to
+                        int varid = varToken->varId();
+                        if (changedVariables.find(varid) != changedVariables.end())
                             continue;
+                        changedVariables.insert(varid);
+                        auto oldValue = data.getValue(varid, nullptr, nullptr);
+                        if (oldValue && oldValue->isUninit())
+                            call(data.callbacks, varToken, oldValue, &data);
+                        if (val->type == ExprEngine::ValueType::ArrayValue) {
+                            // Unknown pointer, array or struct
+                            auto arrayValue = std::dynamic_pointer_cast<ExprEngine::ArrayValue>(val);
+                            assert(arrayValue);
+                            data.assignValue(tok2, varid, val);
```

Comment:
as far as I see this assignment is redundant.. it assigns the same value as we already have.